R/mixture_identification.R

```diff
@@ -37,43 +132,149 @@ mixmod_regression <- function(x, ...) {
 
 #' Mixture Model Identification using Segmented Regression
 #'
-#' @inherit mixmod_regression description details return references
+#' @inherit mixmod_regression description details references
 #'
-#' @inheritParams rank_regression.cdf_estimation
-#' @param control Output of call to \code{\link[segmented]{seg.control}}, which
-#'   is passed to \code{\link[segmented:segmented.lm]{segmented.lm}}.
+#' @inheritParams rank_regression.default
+#' @inheritParams mixmod_regression.cdf_estimation
 #'
-#' @examples
-#' # Data is taken from given reference:
-#' hours <- c(2, 28, 67, 119, 179, 236, 282, 317, 348, 387, 3, 31, 69, 135,
-#'           191, 241, 284, 318, 348, 392, 5, 31, 76, 144, 203, 257, 286,
-#'           320, 350, 412, 8, 52, 78, 157, 211, 261, 298, 327, 360, 446,
-#'           13, 53, 104, 160, 221, 264, 303, 328, 369, 21, 64, 113, 168,
-#'           226, 278, 314, 328, 377)
 #'
-#' status <- c(1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1,
-#'           1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0,
-#'           1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-#'           0, 1, 1, 1, 1, 1, 1)
+#' @return Returns a list of class \code{"rank_regression"} if no breakpoint was
+#' detected. See \code{\link{rank_regression}}. The tibble \code{data} is returned
+#' with class \code{"cdf_estimation"} and contains the additional dummy columns
+#' \code{method} and \code{id}. The first mentioned column is filled with
+#' \code{"_null"}, due to generic visualization functions and the latter is filled
+#' with \code{"XXXXXX"} to point out that unit identification is not possible when
+#' using the vector-based approach.
+#'
+#' Returns a list of class \code{"mixmod_regression"} if at least one breakpoint
+#' was determined. The length of the list depends on the number of identified
+#' subgroups. Each list contains the information provided by
+#' \code{\link{rank_regression}}. The returned tibble \code{data} of
+#' each list element only retains information on the failed units and has modified
+#' and additional columns:
+#' \itemize{
+#'   \item \code{id} : Modified id, overwritten with \code{"XXXXXX"} to point out
+#'     that unit identification is not possible when using the vector-based approach.
+#'   \item \code{method} : A character that is always \code{"_null"}. Due to generic
+#'     visualization functions column \code{method} has to be provided.
```

Comment:
Actually this is not unconditionally true. We could check in visualization functions if data has column method and if not add null value

Reply:
The handle of the vector-based approach in fact needs some refactoring to gain more consistency